### ISSUE TITLE
bgzip output instead of gzip?

### ISSUE THREAD

**Issue by daheise** [poruloh/Eagle#18]
https://github.com/poruloh/Eagle/blob/a10ffd048dfa9a250036bcc899e1f67a888d2ccc/src/FileUtils.cpp#L187

I've had good success with getting performance boosts from [htslib's bgzip parallel](https://github.com/samtools/htslib) compression, it it seems like Eagle would benefit from implementing a bgzip option for output. 

Unfortunately, I'm not familiar enough with C++ streaming operators, nor do I have the bandwidth currently to learn, to contribute a PR.

**Comment by poruloh** [comment]
Eagle's vcf.gz output is actually already in bgzip format and is handled by htslib.

The Oxford haps.gz output format does not currently use bgzip, but given that output compression is a small fraction of Eagle's running time, I don't expect to be able to prioritize adding bgzip to Oxford-format output.

**Comment by daheise** [comment]
Thanks for the information, that's very good to know and helpful for us to get a performance increase in our pipeline. 

Does your bgzip write occur in parallel? If so, could you point me to where in the code? I'm interested in using a htslib parallel bgzip write in a pull request in another project and an example would be very helpful.

**Comment by poruloh** [comment]
I'm not sure whether the bgzip write occurs in parallel. It is implemented simply by calling bcf_write() within the writeVcf() method in Eagle.cpp.

**Comment by daheise** [comment]
Thanks for the specific direction. I have opened a new issue on that topic.
